lib/ash/resource/change/builtins.ex

```diff
@@ -50,16 +58,29 @@ defmodule Ash.Resource.Change.Builtins do
 
     Use `arg(:argument_name)` to use the value of the given argument. If the argument is not supplied then nothing happens.
 
+  ## Additional Options
+
+  #{Spark.OptionsHelpers.docs(@set_attribute_extra_opts)}
+
   ## Examples
 
       change set_attribute(:active, false)
       change set_attribute(:opened_at, &DateTime.utc_now/0)
       change set_attribute(:status, arg(:status))
+      change set_attribute(:encrypted_data, arg(:data), set_when_nil?: false)
   """
-  @spec set_attribute(relationship :: atom, (() -> term) | {:_arg, :status} | term()) ::
+  def set_attribute_extra_opts do
+    @set_attribute_extra_opts
+  end
+
+  @spec set_attribute(relationship :: atom, (() -> term) | {:_arg, :status} | term(), opts :: Keyword.t()) ::
           Ash.Resource.Change.ref()
-  def set_attribute(attribute, value) do
-    {Ash.Resource.Change.SetAttribute, attribute: attribute, value: value}
+  def set_attribute(attribute, value, opts \\ []) do
+    opts =
+      opts
+      |> Keyword.put_new(:set_when_nil?, true)
+
+    {Ash.Resource.Change.SetAttribute, attribute: attribute, value: value, extra_opts: opts}
```

Comment:
I think it would probably be simpler to do`set_when_nil?: Keyword.get(opts, :set_when_nil?, true)` instead of using an `extra_opts` list.

Reply:
I like that plan 👍